core/src/data/mbtilesDataSource.cpp

```diff
@@ -424,51 +434,60 @@ void MBTilesDataSource::initSchema(SQLite::Database& db, std::string _name, std:
 
 bool MBTilesDataSource::getTileData(const TileID& _tileId, std::vector<char>& _data) {
 
-    auto& stmt = m_queries->getTileData;
-    try {
-        // Google TMS to WMTS
-        // https://github.com/mapbox/node-mbtiles/blob/
-        // 4bbfaf991969ce01c31b95184c4f6d5485f717c3/lib/mbtiles.js#L149
-        int z = _tileId.z;
-        int y = (1 << z) - 1 - _tileId.y;
+    int largestLength = 0;
 
-        stmt.bind(1, z);
-        stmt.bind(2, _tileId.x);
-        stmt.bind(3, y);
+    for (int i = 0; i < m_queries.size(); ++i) {
+
+        auto& stmt = m_queries[i]->getTileData;
+        try {
+            // Google TMS to WMTS
+            // https://github.com/mapbox/node-mbtiles/blob/
+            // 4bbfaf991969ce01c31b95184c4f6d5485f717c3/lib/mbtiles.js#L149
+            int z = _tileId.z;
+            int y = (1 << z) - 1 - _tileId.y;
+
+            stmt.bind(1, z);
+            stmt.bind(2, _tileId.x);
+            stmt.bind(3, y);
+
+            if (stmt.executeStep()) {
+                SQLite::Column column = stmt.getColumn(0);
+                const int length = column.getBytes();
 
-        if (stmt.executeStep()) {
-            SQLite::Column column = stmt.getColumn(0);
-            const char* blob = (const char*) column.getBlob();
-            const int length = column.getBytes();
+                if (length > largestLength) {
```

Comment:
Can you add a comment about this check please?

Reply:
If multiple mbtiles contain the same tile, it will get tile which has more data.

Reply:
Sounds great. Can you put this in the code comment please, will be helpful for anyone looking at the code in future. Thanks

Reply:
Ok, I have added it.

Reply:
Thanks @rwrx